src/browser-polyfill.js

```diff
@@ -440,6 +465,31 @@ if (typeof browser === "undefined" || Object.getPrototypeOf(browser) !== Object.
       };
     });
 
+    /**
+     * Creates and returns a function which, when called, will resolve or reject
+     * the given promise based on how it is called:
+     *
+     * - If, when called, `chrome.runtime.lastError` contains a non-null object,
+     *   the promise is rejected with that value.
+     * - If the function is called with exactly one argument, the promise is
+     *   resolved to that value.
+     * - Otherwise, the promise is resolved to an array containing all of the
+     *   function's arguments.
+     *
+     * @param {object} promise
+     *        An object containing the resolution and rejection functions of a
+     *        promise.
+     * @param {function(any):void} promise.resolve
+     *        The promise's resolution function.
+     * @param {function(any):void} promise.reject
```

Comment:
I'm aware that these names are likely inspired by the current naming of `makeCallback`, but the name is misleading. Please rename `promise` to `deferred` (since elsewhere we already use the `promise` name for actual promises).

If you prefer, you can leave the current name as it is and make a separate PR to have consistent naming in the whole file.

Reply:
Let’s do that in a&nbsp;latter PR.